modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/jaxrs/writer/interceptor/NestedFieldsWriterInterceptor.java

```diff
@@ -481,6 +486,14 @@ private void _resetNestedAwareMessage(Message message) {
 			message.put(
 				"HTTP.REQUEST",
 				nestedFieldsHttpServletRequestWrapper.getRequest());
+
+			Exchange exchange = message.getExchange();
+
+			if (exchange instanceof ExchangeWrapper) {
```

Comment:
Here, instead of setting the resource to null, you could just get rid of the wrapper:

```
Exchange exchange = message.getExchange();

if (exchange instanceof ExchangeWrapper) {
	ExchangeWrapper exchangeWrapper = (ExchangeWrapper)exchange;

	message.setExchange(exchangeWrapper.getExchange());
}
```

Reply:
Done! Great idea!